### PR TITLE
Comments galore. 

### PR DESCRIPTION
**Pull request by ccjeane**
- Created a comment class that stores ID's and Voters
- Users can only vote once on a review!
- PlaceReview object used to transfer data between Java and Javascript
- Style Fixes
- Recent Review Sorting

### CODE REVIEW

File: src/main/java/com/google/sps/data/PlaceReviews.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  /**
+   * Accessor Method
+   * Searches the existing Array of comments to find the other by the same author
+   * Assumes that there was a previous post by author
+   * @return Comment the previous comment
+   *                  TODO: Implement with Binary Search
+   */
+   private Comment getPrevReview(String author) {
+    Collections.sort(this.reviews, Comment.ORDER_BY_AUTHOR);
+     for (Comment cur : this.reviews) {
```
Comment (mwo-2020) [inline src/main/java/com/google/sps/data/PlaceReviews.java:79]: Indent.

Why is sorting needed here before a linear search?

Comment (ccjeane) [reply src/main/java/com/google/sps/data/PlaceReviews.java:79]: Not sure, will loop in @dterr

Comment (dterr) [reply src/main/java/com/google/sps/data/PlaceReviews.java:79]: I wrote in the sort by author as an attempt to implement a binary search, which I ended up not implementing. This is poor style for runtime and I will change it.

Comment (dterr) [reply src/main/java/com/google/sps/data/PlaceReviews.java:79]: #99

Comment (ccjeane) [reply src/main/java/com/google/sps/data/PlaceReviews.java:79]: Awesome - thanks for quickly fixing that!

File: src/main/java/com/google/sps/data/PlaceReviews.java [hunk 2]
```diff
@@ -73,9 +73,9 @@ private void replaceReview(Comment review) {
    * Assumes that there was a previous post by author
    * @return Comment the previous comment
    *                  TODO: Implement with Binary Search
+   *                        Collections.sort(this.reviews, Comment.ORDER_BY_AUTHOR);
```
Comment (fyordan) [inline src/main/java/com/google/sps/data/PlaceReviews.java]: TODO has a lot of indent, remove unneccesary whitespace. Also, given time constraints, and low number of reviews, might not be worth doing binary search.

**Comment by fyordan** [comment]
Is this tested and ready for submitting? Branch PR is getting large.

**Comment by ccjeane** [comment]
Yes! Everything compiles and works.

File: src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -1,7 +1,5 @@
 package com.google.sps.data;
-
-import java.util.Date;
-import java.util.Comparator;
+import java.util.*;
```
Comment (fyordan) [inline src/main/java/com/google/sps/data/Comment.java:2]: Please use only the ones you need. Importing unneccessary libraries can make the binary larger which in turn makes it take more space and longer to build.

Comment (ccjeane) [reply src/main/java/com/google/sps/data/Comment.java:2]: Got it - just pushed out the PR fix!

File: src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  };
+  
+  /** 
+   * Constructor
+   * Creates a full Comment object, used when an author leaves a review for the first time
+   * @param author the message author
+   * @param message the content of the message
+   * @param timestamp the time of writing
+   */
+  public Comment(String author, String message, Date timestamp, Long pos, Long neg) {
```
Comment (kcirerick) [inline src/main/java/com/google/sps/data/Comment.java:60]: consider moving constructor above comparator classes

File: src/main/java/com/google/sps/data/Comment.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    this.negative = neg;
+    this.voters = new HashSet<>();
+  }
+
+  /* Add a voter to the voters set. */
+  public void addVoter(String voter){
+    voters.add(voter);
+  }
+
+  public long getId(){
```
Comment (kcirerick) [inline src/main/java/com/google/sps/data/Comment.java:74]: comments? here and below

File: src/main/java/com/google/sps/data/PlaceReviews.java (hunk 1 above)
Comment (kcirerick) [reply src/main/java/com/google/sps/data/PlaceReviews.java:79]: do we still need the todo or as this been resolved?

File: src/main/java/com/google/sps/data/Comment.java (hunk 4 above)
Comment (ccjeane) [reply src/main/java/com/google/sps/data/Comment.java:63]: @dterr

Comment (ccjeane) [reply src/main/java/com/google/sps/data/Comment.java:63]: Also, why?

File: src/main/java/com/google/sps/data/PlaceReviews.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    }
+  }
+
+  /**
+   * Updater method
+   * Replaces previous review with the new review for a given author
+   */
+   private void replaceReview(Comment review) {
+    Comment prevReview = getPrevReview(review.getAuthor());
+    prevReview.updateComment(review);
```
Comment (fyordan) [inline src/main/java/com/google/sps/data/PlaceReviews.java:70]: Indentation

**Pull request merged**